src/inventory_ui.cpp

```diff
@@ -910,7 +921,22 @@ item_location game::inv_map_splice(
                     }
                 }
             }
-            inv_s.make_item_list( slices.back(), pos == g->u.pos() ? &ground_cat : &nearby_cat );
+            if ( found_item ) {
+                if( pos == g->u.pos() ) {
+                    inv_s.make_item_list( slices.back(), &ground_cat );
+                } else {
+                    int dist = square_dist( g->u.pos(), pos );
+                    std::string dir_name = direction_name_short( direction_from( g->u.pos(), pos ) );
```

Comment:
Consider factoring this as a separate function so it can be easily extended. You could implement `map::name_tile( const player *p)` or even better extend `item_location::name(const player *p)`